### PR TITLE
Adds energy bola

### PR DESCRIPTION
**Pull request by SpeltIncorrectyl**
## About the PR
Energy bola is a tool available to security. It is like a normal bola except it disintegrates when removed, sort of like zipties.

## Why / Balance
The energy bola has identical stats to the normal bola, except it takes up half the inventory space (2 slots vs 4 slots of the bola) and is single use.
There are 2 energy bolas in every security officer locker, 2 in each in HoS locker and Warden locker, 8 in the sectech, and they can be manufactured at the secfab for 3 plastic each.

## Technical details
Added new property to EnsaringCompnent, DestroyOnRemove. When something with this set to true is removed from the victim, it is deleted instead of being placed into someone's hand.

## Media

https://github.com/user-attachments/assets/428a9487-372b-41e0-8cfd-2e25e72616dd

## Requirements

- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
* Potential breaking change, new property DestroyOnRemove added to EnsnaringComponent, defaults to false which performs original behaviour.

**Changelog**
:cl:
- add: Added energy bolas to the security department's arsenal. They are like normal bolas except twice as cool.

### CODE REVIEW

File: Content.Shared/Ensnaring/Components/EnsnaringComponent.cs
Comment (slarticodefast) [inline Content.Shared/Ensnaring/Components/EnsnaringComponent.cs:68]:
```suggestion
    [DataField]
```
ReadWrite is the default now, so you don't need this attribute. You can also remove it from the other datafields above. Change them to `[DataField]` as well. The name gets autoconverted to camelCase, but old files still do this manually. And add a summary to your new datafield.

**Review by slarticodefast — CHANGES_REQUESTED** [review]
Very minor cleanup needed, but looks good to me code-wise. From a balancing perspective I will put it up for discussion with the other maintainers.

**Review by slarticodefast — APPROVED** [review]

**Comment by whateverusername0** [comment]
i love ebola, merge now

**Comment by SpeltIncorrectyl** [comment]
guh
another merge conflict!?

**Comment by slarticodefast** [comment]
I had a balance discussion with the other maintainers. Conceptually this addition has our approval, but we think in in its current form it might be a very powerful weapon, in particular due to the self-deletion. We would ask you to make the following adjustments:
- Make its size 2x2, the same as the normal bola.
- Make it unlockable by science research instead of it being available for mass production at round start.
- Remove the energy bolas from the lockers so that its round start availability is reduced. Keeping a few in the sec vend is fine.
- Reduce the breakoutTime to something like 1-2 seconds (and adjust the freeTime accordingly).
- Add comments in the yml code stating that it is balanced around its self deleting ability and low availability (so future balance PRs will know our reasoning on that).

**Comment by SpeltIncorrectyl** [comment]
> I had a balance discussion with the other maintainers. Conceptually this addition has our approval, but we think in in its current form it might be a very powerful weapon, in particular due to the self-deletion. We would ask you to make the following adjustments:
> 
>     * Make its size 2x2, the same as the normal bola.
> 
>     * Make it unlockable by science research instead of it being available for mass production at round start.
> 
>     * Remove the energy bolas from the lockers so that its round start availability is reduced. Keeping a few in the sec vend is fine.
> 
>     * Reduce the breakoutTime to something like 1-2 seconds (and adjust the freeTime accordingly).
> 
>     * Add comments in the yml code stating that it is balanced around its self deleting ability and low availability (so future balance PRs will know our reasoning on that).

If it just didn't have self-deletion could it keep its size and availability?

**Comment by UbaserB** [comment]
that'd pretty much be normal bolas at that point lol

**Comment by slarticodefast** [comment]
> that'd pretty much be normal bolas at that point lol

Yeah, this pretty much. There was no opposition to sec having bolas in general though.

**Comment by SpeltIncorrectyl** [comment]
> > that'd pretty much be normal bolas at that point lol
> 
> Yeah, this pretty much. There was no opposition to sec having bolas in general though.

Could energy bola be a smaller cheaper bola for sec?

**Comment by SpeltIncorrectyl** [comment]
I've done the changes that Keron said to in design-discussion on the discord.
* 2x2 size in inventory to match normal bola
* 5000 point tier 1 arsenal research to make them in the secfab
* no self deletion

File: Resources/Prototypes/Entities/Objects/Weapons/Throwable/bola.yml
Comment (slarticodefast) [inline Resources/Prototypes/Entities/Objects/Weapons/Throwable/bola.yml:73]:
```suggestion
```
This part is not needed since it is inherited from BaseBola

**Review by slarticodefast — CHANGES_REQUESTED** [review]

**Comment by chavonadelal** [comment]
I think you shouldn't separate this out as a separate technology. You should put it in the technology special means or counteraction to riots (I think that's what it's called). Otherwise it's very cool!

**Review by chavonadelal — CHANGES_REQUESTED** [review]
I think you shouldn't separate this out as a separate technology. You should put it in the technology special means or counteraction to riots (I think that's what it's called). Otherwise it's very cool!

**Comment by slarticodefast** [comment]
Hey, we had a discussion and vote among the maintainers on if and how the energy bola should be merged. The poll options and results were:
- Merge as it is, the same as the normal bola, but with heat damage (1)
- Change it to be self-deleting again, then merge (3)
- Give it an alternative ability, different from self-deletion (0)
- Close the PR (1)

Since the required 75% approval was not reached, the next step according to the [maintainer policy](https://docs.spacestation14.com/en/staff/maintainer-policy.html) will be further discussion about possible adjustments before a second poll can be made. If you have any suggestions for this, feel free to tell us either here or in the #design-discussion channel on Discord. The main points made in the previous discussion were the following:
- The energy bola should have different stats/properties than the normal bola, enough to warrant it being its own item
- The self-deletion property was extremely powerful in SS13, raising concerns for our combat balance.
- In all cases the round start availability should be greatly reduced. With two in each locker and 8 in the sec vend on a map like Marathon you would have 28 in total. That is way more than you would ever need and makes the research and printing completely unnecessary. I would suggest only keeping a few in the sec vend.
- The inventory size should be the same as the normal bola, since you should not be able to carry dozens of it with you.
- The energy bola should not be its own research, but be part of an existing tech, for example riot control.
- One maintainer voiced concerns about the item sprite, but did not take part in the poll.

**Comment by SpeltIncorrectyl** [comment]
Is this pr still under review?

**Comment by Plykiya** [comment]
> Is this pr still under review?

the wall of text you received was your review, make the changes they asked for and it'll be aporoved

**Comment by slarticodefast** [comment]
> the wall of text you received in the previous message was your review, make the changes they asked for and it'll be re-reviewed

Yeah, this. When some adjustments are made we can do a second poll among maintainers, but we are currently unsure ourselves about how to best balance this item while keeping it unique. Suggestions are welcome.

**Comment by SpeltIncorrectyl** [comment]
I didn't know how to make it a unique item without self-deletion or a different size.

**Pull request closed without merging**
